expo/expo
# [docs] Update APISection test, regenerate expo-apple-authentication data

# Why



![CleanShot 2024-11-15 at 13 15 21@2x](https://github.com/user-attachments/assets/8607eb2c-3a24-4b0a-bb37-5e0d984c052d)


![CleanShot 2024-11-15 at 15 41 19@2x](https://github.com/user-attachments/assets/c29aa469-41cd-4d8f-b1db-85c9bfc07bed)


# How



- Fix broken internal links in `expo-apple-authentication` and generate JSON data file because failing snapshot test depends on it.
- Update `APISection.test.tsx` to have headings length to 26 and tables to 12.

# Test Plan



- Run `yarn build` from `expo-apple-authentication` package
- Run `et gad -p expo-apple-authentication`
- Inside `docs` directory, run `yarn run test -u` and it should update the snapshot test with updated value

# Checklist
